Restructure docs (#254)
Quansight/qhub: docs/docs/step-by-step-walkthrough.md

# Step-by-Step QHub Cloud Deployment

> The list above is our suggestion as other providers can be used. You will need to consult their documentation on setting up oauth and DNS registry.

Author: ```suggestion
The list below is our suggestion. Other providers may be used, but you will need to consult their documentation. When using a different provider the corresponding flag in the `qhub` command-line argument should be omitted.
```

> If a different provider is desired, then the corresponding flag in the `qhub` command line argument can be omitted.

Author: ```suggestion
```

For this initial tutorial, we assume that you will use the following tools:

- [Github actions] will be used for CI/CD
- Oauth will be via [auth0]
- DNS registry will be through [Cloudflare]

## 1. Install QHub
Following best practice, we recommend that you create a virtual environment previously and from there install QHub on machine:

- If you prefer conda, install the latest stable release with : 
```bash
    conda install -c conda-forge qhub
```

- If you are a PyPI user instead, download the latest stable release with: 
```bash
    pip install qhub
```
    
- (For developers) The latest development version can be installed using PyPI: 
```bash
    pip install git+https://github.com/Quansight/qhub.git
```
> NOTE: Download the 'development version' to try out features which have not been yet officially released. 
    

## 2. Environment variables

In order to fully automate the deployment, several environment variables must be set. The following subsections will 

> describe the purpose and method for obtaining the values for these variables.

Author: ```suggestion
describe the purpose and method for obtaining the values for these variables. An introduction to unix environment variables can be found [here](https://en.wikipedia.org/wiki/Environment_variable).
```

Reviewer: ```suggestion
describe the purpose and method for obtaining the values for these variables. An introduction to UNIX environment variables can be found [here](https://en.wikipedia.org/wiki/Environment_variable).
```

>     git push origin master

Author: ```suggestion
    git push origin main
```

## 3. Post GitHub deployment:

> After the files are in github all CI/CD changes will be done via github actions and triggered by a commit to master. To use gitops, make a change to the `qhub-ops.yaml` in a new branch and create pull request into master. When the pull request is merged, it will trigger a deployement of all of those changes to your QHub.

Author: ```suggestion
After the files are in Github all CI/CD changes will be triggered by a commit to main and deployed via GitHub actions. To use gitops, make a change to `qhub-ops.yaml` in a new branch and create a pull request into main. When the pull request is merged, it will trigger a deployment of all of those changes to your QHub.
```
